cmd/kubelet/app/options/options.go

```diff
@@ -366,7 +366,7 @@ func (f *KubeletFlags) AddFlags(mainfs *pflag.FlagSet) {
 	fs.StringVar(&f.CertDirectory, "cert-dir", f.CertDirectory, "The directory where the TLS certs are located. "+
 		"If --tls-cert-file and --tls-private-key-file are provided, this flag will be ignored.")
 
-	fs.StringVar(&f.CloudProvider, "cloud-provider", f.CloudProvider, "The provider for cloud services. Specify empty string for running with no cloud provider.")
+	fs.StringVar(&f.CloudProvider, "cloud-provider", f.CloudProvider, "The provider for cloud services. Specify empty string for running with no cloud provider. If set, the cloud provider determines the name of the node (consult cloud provider documentation to determine if and how the hostname is used).")
```

Comment:
> If set, the cloud provider determines the **name** of the node 

That's not true is it? The cloud provider will only set the address of type hostname; not change the nodes name.

Reply:
The cloud provider does determine the node name. Some use the detected or overridden hostname to do so, some don't.

Reply:
See https://github.com/kubernetes/website/pull/8873 describes how each cloud provider determines the node name.